### ISSUE TITLE
Could not load the file sensor_msgs/JointState.h

### ISSUE THREAD

**Issue by r-hede** [ros/robot_state_publisher#110]
Hi,

I have an error with the package `robot_state_publisher` when I'm try to `catkin_make_isolated` with `ROS Melodic` on `Raspbian 9 (stretch)`.

Specifically, I get an error from the file `robot_state_publisher/include/robot_state_publisher/joint_state_listener.h` on line `43`.

`fatal error: sensor_msgs/JointState.h: No such file or directory`

However, the dependency `sensor_msgs` is correctly loaded in the `package.xml` file and the searched file exists at the indicated path.

Could you help me ?

Best regards,

Robin Hède.

**Comment by sloretz** [comment]
It sounds like `sensor_msgs` might have been built in another workspace that was not sourced. It will be hard to tell without steps to reproduce. Adding ` -DCMAKE_VERBOSE_MAKEFILE:BOOL=ON` to your `catkin_make_isolated` call might help you debug which include paths are being passed.

This might fit better as a question on https://answers.ros.org . I'll leave this issue open for a bit in case you discover a bug in `robot_state_publisher`.

**Comment by r-hede** [comment]
I finally solved my problem by using another package path.

Instead of using the "ros_comm" package path, I used "robot".

`rosinstall_generator robot --rosdistro melodic --deps --tar > melodic-robot.rosinstall`

The [robot variant](http://www.ros.org/reps/rep-0131.html#robot-variant) is defined to be core, stable, ROS libraries for any robot hardware. It is the "general robotics" libraries of ROS. It may not contain any GUI dependencies.

The complete procedure to installing the Melodic version of ROS from sources is available [here](https://wiki.ros.org/melodic/Installation/Source).
